core/src/main/kotlin/org/evomaster/core/search/service/monitor/SearchProcessMonitor.kt

```diff
@@ -0,0 +1,189 @@
+package org.evomaster.exps.monitor
+
+import com.google.gson.GsonBuilder
+import com.google.inject.Inject
+import com.sun.org.apache.xpath.internal.operations.Bool
+import org.evomaster.core.EMConfig
+import org.evomaster.core.problem.rest.RestAction
+import org.evomaster.core.problem.rest.param.Param
+import org.evomaster.core.search.EvaluatedIndividual
+import org.evomaster.core.search.Individual
+import org.evomaster.core.search.gene.Gene
+import org.evomaster.core.search.service.*
+import java.io.File
+import java.io.IOException
+import java.lang.IllegalStateException
+import java.nio.ByteBuffer
+import java.nio.channels.FileChannel
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption
+import javax.annotation.PostConstruct
+
+
+/**
+ * @author: manzhang
+ * @date: 31/08/2018
+ * @decription: monitor 1) how are rest actions or rest individual selected regarding how to sampleAll
+ *                      2) how does targets update
+ */
+
+
+class SearchProcessMonitor: SearchListener {
+
+    @Inject
+    private lateinit var time: SearchTimeController
+
+    @Inject
+    private lateinit var config: EMConfig
+
+    @Inject
+    private lateinit var archive: Archive<*>
+
+    @Inject
+    private lateinit var idMapper: IdMapper
+
+    private lateinit var overall : SearchOverall<*>
+
+    var eval: EvaluatedIndividual<*>? = null
+    var step :StepOfSearchProcess<*>? = null
+    var isMutated : Boolean = false
+
+    private var cs = ""
```

Comment:
"cs" and "tb" do not really say what they are for. short names are fine for local variables in functions, but usually not fields

Reply:
cs does not make sense now.  I removed this now